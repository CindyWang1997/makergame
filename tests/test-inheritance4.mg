object parent {
  void compute() { std::printstr("parent compute"); }
  event create { std::printstr("parent create"); }
}

object child : parent {
  void compute() { std::printstr("child compute"); }
  event create { std::printstr("child create"); } 
}

object main {
  int i;
  event create {
    child c = create child; // child create
    c.compute();            // child compute

    parent p = create parent; // parent create
    p.compute();              // parent compute

    p = c;
    p.compute(); // parent compute, despite pointing to child, since non-virtual
    std::end_game();
  }
}
