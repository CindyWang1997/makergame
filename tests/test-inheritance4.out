child create
child compute
parent create
parent compute
parent compute
